pkg/operation/common/extensions.go

```diff
@@ -335,10 +344,15 @@ func WaitUntilExtensionCRDeleted(
 	name string,
 	interval time.Duration,
 	timeout time.Duration,
+	waiter retry.Ops,
```

Comment:
Now we pass `interval` + `timeout` + `waiter` separately even though they belong together. Additionally, I think the parameter count and passing `nil` so many times gets a little awkward. I thought about two possible ways:
1. Passing a `WaitFn` instead of `interval` + `timeout` + `waiter`. Helper functions could help to create a proper `WaitFn`, i.e. one for tests and one for prod execution.
2. Controlling the `waiter` via a global variable and exchanging it before executing the tests. 

If the "default waiter" is anyway problematic for the test execution, it shouldn't harm to exchange it globally.

WDYT?

Reply:
Well, that's a valid point. I like the second approach as it is more lightweight and doesn't introduce more complexity and helper/wrapper functions.

Then let's simply exchange `retry.{Until,UntilTimeout}` with the fake implementations in the tests where we need it (these are anyways aliases already). Would that sound good to you?

Reply:
Sure :)